dart/lib/flat_buffers.dart

```diff
@@ -1165,8 +1173,12 @@ class Uint16ListReader extends Reader<List<int>> {
 
   @override
   @pragma('vm:prefer-inline')
-  List<int> read(BufferContext bc, int offset) =>
-      _FbUint16List(bc, bc.derefObject(offset));
+  List<int> read(BufferContext bc, int offset) {
+    final listOffset = bc.derefObject(offset);
+    final length = bc._getUint32(listOffset);
+
+    return bc._asUint16List(listOffset + _sizeofUint32, length);
```

Comment:
This only works on little-endian hosts. While the vast majority would be little-endian, which is what FlatBuffers uses, it's possible to run Dart on big-endian.

How about we kept the original private classes as a fallback? 
```suggestion
    if (Endian.host == Endian.little) {
        final listOffset = bc.derefObject(offset);
        final length = bc._getUint32(listOffset);
        return bc._asUint16List(listOffset + _sizeofUint32, length);
    } else {
        _FbUint16List(bc, bc.derefObject(offset));
    }
```

Reply:
@vaind Okay, but what do you suggest in the case of Uint8ListReader which has a lazy path? Are you suggesting that the following change be made? The greedy path would remain unchanged.

```dart
List<int> read(BufferContext bc, int offset) {
  final listOffset = bc.derefObject(offset);
  final length = bc._getUint32(listOffset);

  if (lazy) {
    if (Endian.host == Endian.big) {
      return _FbUint8List(bc, listOffset);
    } else {
      return bc._asUint8List(listOffset + _sizeofUint32, length);
    }
  }

  final result = Uint8List(length);
  var pos = listOffset + _sizeofUint32;
  for (var i = 0; i < length; i++, pos++) {
    result[i] = bc._getUint8(pos);
  }
  return result;
}
```

Reply:
uint8list is not affected by endianness because each item is just a single byte

Reply:
You know, I should have finished my morning coffee before writing comments. I don't even know how I processed that one. Lmao. Will push a follow-up commit shortly.